platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -603,17 +610,21 @@ public VirtualLeafRecord<K, V> loadLeafRecord(final K key) throws IOException {
     }
 
     /**
-     * Load a leaf record by path
+     * Load a leaf record by path. This method returns {@code null}, if the path is outside the
+     * valid path range.
      *
      * @param path the path for the leaf we are loading
      * @return loaded record or null if not found
      * @throws IOException If there was a problem reading record from db
      */
     @Override
```

Comment:
This method needs `@Nullable` annotation, according to the description

Reply:
OK, I'm adding `@Nullable` to `loadLeafRecord()`, `loadHash()`, and `findKey()` methods - all of them may return null values